api/src/api/feature_flags/feature_flag.py

```diff
@@ -0,0 +1,28 @@
+from enum import StrEnum
+
+
+class FeatureFlag(StrEnum):
+    """
+    This enum class serves as a list of constant values for feature flags.
+
+    A value like:
+        EXAMPLE_FLAG_A = "example_flag_a"
+
+    Would be able to have its default value set by setting the "EXAMPLE_FLAG_A" environment
+    variable, and be overrideable in an endpoint by supplying it as "X-FF-Example-Flag-A" in a header.
+
+    """
+
+    ### NOTE: This is a placeholder for future work, and only serves as a proof of concept of the idea.
+    #
+    # Header: X-FF-Enable-Opportunity-Log-Msg
+    # EnvVar: ENABLE_OPPORTUNITY_LOG_MSG
+    ENABLE_OPPORTUNITY_LOG_MSG = "enable_opportunity_log_msg"
+
+    def get_header_name(self) -> str:
+        value = "-".join([v.capitalize() for v in self.value.lower().split("_")])
```

Comment:
Not clear, but aren't headers case-insensitive? What's the point of the funky capitalization + this logic to handle it?

Reply:
Mainly so the docs generate with the normal convention